HW06.Rmd

```diff
@@ -1,92 +1,115 @@
-
-
-GBM.genes.of.interest <- filter(ens.mapped_GBM, gene_symbol %in% genes_of_interest)
-
-#Now perform a t-test between the H3K mutated and the wt samples. There are many ways to do this actually, you can use a for loop or you could do the tidy alternative with broom(), but the for loop is probably the easiest
-
-#print out the t-test results
-```
-
-
-
-Now create a graphing function to create boxplots to visualize the results. Plot expression on the y-axis. The graph should look like [this example](IRX1_example.png)
-
-```{r create a graphing function}
-#to work in the tidyverse, it will be easier to make tidy the dataframe first
-
-
-
-#create a graphing function
-
-#then use a for loop combined with the graphing function to make a graph for all your genes of interest 
-```
-
-
-```{r reproducibility}
-sessionInfo()
-```
-
+---
+title: "HW06"
+output: github_document
+---
+
+In cancer, mutations arise that promote growth or survival of cells. In glioblastoma multiforme and other high grade gliomas, a common mutation is a mutation of the 27th lysine (K) to a methionine (M) of the histone subunit H3, or in short H3K27M. 
+
+H3K27M is the most frequent oncohistone in brain cancers, but the biology is still not well understood. Your analysis is to look at the expression of several (27) genes to see if they are differentially expressed and plot 27 boxplots each gene. The data used in this analysis was obtained [from this publication](https://pubmed.ncbi.nlm.nih.gov/31085178/) 
+
+Steps: 
+
+  1. Read in the 45 processed RNA-Seq data found in "./RNA_Seq_processed"
+  2. Map gene.ids to gene.symbols (which I have for you)
+  3. For 27 genes of interest AND your favorite gene, perform a t-test to see if it is differentially expressed between the WT vs H3K27M samples
+  4. Create a graphing function and then create a boxplot that graphs expression between the two groups
+  
+  
+#### Code
+
+From the RNA-Seq files, you only need the "Name" from one file and the "TPM" column from all the files. TPM stands for "transcripts per million" and is a common unit for normalized expression data. 
+```{r load in data, message=FALSE, warning=FALSE}
+#hint, using apply (specifically sapply) you can read in the data into a list and then bind the columns together. Or you can use a for loop too. 
+
+library(magrittr)
+library(knitr)
+library(tidyverse)
+library(tidymodels)
+
+gene_ids <- read.csv("RNA_Seq_processed/WT/1840.sf", sep = "\t", header = TRUE)$Name
+GBM.transcripts <- list.files("RNA_Seq_processed/", full.names = TRUE, recursive = TRUE) %>% 
+  sapply(function(f) {read.csv(f, sep = '\t', header = TRUE)$TPM}) %>% as.data.frame()
+colnames(GBM.transcripts) <- list.files("RNA_Seq_processed/", recursive = TRUE) %>% 
+  strsplit('[\\.]') %>% sapply(function(x) {x[1]})    # Piping can avoid ugly long function nestings
+GBM.transcripts$gene_id <- gene_ids
```

Comment:
This chunk seems to work well. I tested it out on my own and it seems to meet the prompt with very few lines.  I think the choice to create a function within the sapply was good as it was only used once.

Reply:
I agree this snippet could be a bit confusing. This is definitely not the kind of everyday code - just to see how far I can compress my them. Only line 31 is extracting the real (numeric) data. For line 32-34, I tried to attach reasonable column names and gene IDs to the data chunk. I should have them commented more appropriately.